addon/lib/dependencies.js

```diff
@@ -0,0 +1,134 @@
+import { assert } from "@ember/debug";
+import { computed, get } from "@ember/object";
+
+import { getAST, getTransforms } from "ember-caluma/utils/jexl";
+
+/**
+ * Extract dependency slugs from a jexl expression.
+ *
+ * E.g: 'foo'|answer in 'bar'|answer|mapby('column') would result in the
+ * following dependency slugs: ['foo', 'bar', 'bar.column']
+ *
+ * @param {Jexl} jexl
+ * @param {String} expression
+ * @return {String[]} Returns an array of dependency slugs
+ */
+export function getDependenciesFromJexl(jexl, expression) {
+  const allTransforms = getTransforms(getAST(jexl, expression));
+  const answerTransforms = allTransforms.filter(
+    (transform) => transform.name === "answer"
+  );
+  const mapbyTransforms = allTransforms.filter(
+    (transform) =>
+      transform.name === "mapby" &&
+      transform.args[0].type === "FunctionCall" &&
+      transform.args[0].name === "answer"
```

Comment:
Should we also make sure that `transform.args[0].args[0].type == "Literal"`? Otherwise e.g. `bar|answer|mapby('column')` would cause a dependency to `bar` to be listed. Since we don't support this we might want to throw an error or a warning in that case?

Reply:
If we decide to do that, we should do the same for the `answer` transform.

Reply:
That can't happen since the backend validates this..

Reply:
:+1: Alright